python_modules/dagster/dagster/_core/definitions/declarative_automation/operands/code_version_changed_condition.py

```diff
@@ -1,19 +1,14 @@
-from typing import Optional
-
+from dagster._core.definitions.asset_key import AssetKey
 from dagster._core.definitions.declarative_automation.automation_condition import (
-    AutomationCondition,
     AutomationResult,
+    BuiltinAutomationCondition,
 )
 from dagster._core.definitions.declarative_automation.automation_context import AutomationContext
-from dagster._record import record
 from dagster._serdes.serdes import whitelist_for_serdes
 
 
 @whitelist_for_serdes
-@record
-class CodeVersionChangedCondition(AutomationCondition):
-    label: Optional[str] = None
-
+class CodeVersionChangedCondition(BuiltinAutomationCondition[AssetKey]):
```

Comment:
can this not apply to any entity?

Reply:
asset checks do not currently have code versions, although I don't think there's any particular reason they couldn't